### PR TITLE
Add precursor_mz harmonization and make sure losses are calculated.

### LINKED ISSUE #211: Update pretrained test word2vec model to include losses.
The ``add_losses`` filter was calling ``spectrum.get("precursor_mz")`` which so far didn't exist. That essentially means that the test models so far have been trained without losses.

### PR DESCRIPTION
**Pull request by florian-huber**
Addressing #211 

- Added ``add_precursor_mz`` function.
- Retrained a test model and updates the integration tests.

### CODE REVIEW

File: integration-tests/test_user_workflow_spec2vec.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def apply_my_filters(s):
         s = require_minimum_number_of_peaks(s, n_required=5)
         return s
 
-    repository_root = os.path.join(os.path.dirname(__file__), "..")
+    repository_root = os.path.join(os.path.dirname(__file__), '..')
```
Comment (fdiblen) [inline integration-tests/test_user_workflow_spec2vec.py]: use double quotes

File: integration-tests/test_user_workflow_spec2vec.py [hunk 2]
```diff
@@ -46,6 +46,9 @@ def apply_my_filters(s):
         model.train([d.words for d in documents], total_examples=len(documents), epochs=20)
         model.save(model_file)
 
+    assert "loss@66.1" in model.wv.vocab, 'Word should be in corpus and model.'
```
Comment (fdiblen) [inline integration-tests/test_user_workflow_spec2vec.py]: add to a separate test

File: matchms/filtering/harmonize_precursor_mz.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        return None
+
+    spectrum = spectrum_in.clone()
+
+    if isinstance(spectrum.get("precursor_mz", None), str):
+        spectrum.set("precursor_mz", float(spectrum.get("precursor_mz").strip()))
+    elif spectrum.get("precursor_mz", None) is None:
+        pepmass = spectrum.get("pepmass")
+        if pepmass:
+            spectrum.set("precursor_mz", pepmass[0])
```
Comment (fdiblen) [inline matchms/filtering/harmonize_precursor_mz.py]: add a test to at least check the type

**Review by fdiblen — CHANGES_REQUESTED** [review]
Cool. requested few changes.

File: integration-tests/test_user_workflow_spec2vec.py (hunk 1 above)
Comment (florian-huber) [reply integration-tests/test_user_workflow_spec2vec.py]: Done.

File: integration-tests/test_user_workflow_spec2vec.py (hunk 2 above)
Comment (florian-huber) [reply integration-tests/test_user_workflow_spec2vec.py]: Removed this. 
Will add it as a new test in a next PR (which will deal more in depth with the ``add_losses`` function.

File: matchms/filtering/harmonize_precursor_mz.py (hunk 3 above)
Comment (florian-huber) [reply matchms/filtering/harmonize_precursor_mz.py]: Done.

**Comment by florian-huber** [comment]
@fdiblen Thanks for reviewing! 
I added changes as response to your comments. Let me know if they are resolved so I can merge this PR.

**Review by fdiblen — APPROVED** [review]
:swan:

**Comment by florian-huber** [comment]
Thanks Faruk.

**Pull request merged**